deps/wazuh_testing/wazuh_testing/vulnerability_detector.py

```diff
@@ -321,3 +368,291 @@ def modify_system(os_name="CentOS Linux", os_major="7", name="centos7", agent_id
+        Number of expected vulnerabilities
+    feed_source : str
+        OVAL or NVD
+    update_position : boolean
+        Filter configuration parameter to search in Wazuh log
+    timeout : str
+        Timeout to check the event in Wazuh log
+    """
+    check_vuln_detector_event(
+        wazuh_log_monitor=wazuh_log_monitor, update_position=update_position, timeout=timeout,
+        callback=f"The {feed_source} found a total of '{expected_vulnerabilities_number}' potential vulnerabilities "
+                 "for agent .*",
+        error_message=f"The expected number of {feed_source} vulnerabilities have not been found"
+    )
+
+
+def check_package_vulnerable_report(wazuh_log_monitor, package, cve, update_position=False,
+                                    timeout=VULN_DETECTOR_GLOBAL_TIMEOUT):
+    """
+    Check if the package vulnerable has been reported
+
+    Parameters
+    ----------
+    wazuh_log_monitor: FileMonitor
+        FileMonitor object to monitor the Wazuh log
+    package : str
+        Package name to check
+    cve : str
+        CVE to check
+    update_position : boolean
+        Filter configuration parameter to search in Wazuh log
+    timeout : str
+        Timeout to check the event in Wazuh log
+    """
+    check_vuln_detector_event(
+        wazuh_log_monitor=wazuh_log_monitor, update_position=update_position, timeout=timeout,
+        callback=f"The '{package}' package .* from agent .* is vulnerable to '{cve}'",
+        error_message=f"Could not find the report which says that the package {package} is vulnerable with {cve}")
+
+
+def set_system(system):
+    """
+    Set custom system in global DB Agent info
+
+    Parameters
+    ----------
+    system: str
+        System to set. Available systems in SYSTEM_DATA variable
+    """
+    try:
+        set_custom_system(os_name=SYSTEM_DATA[system]['os_name'], os_major=SYSTEM_DATA[system]['os_major'],
+                          os_minor=SYSTEM_DATA[system]['os_minor'], name=SYSTEM_DATA[system]['name'])
+    except KeyError:
+        pass
+
+
+def insert_data_redhat_feed(data, field_name, field_value, append_data):
+    """
+    Allow insert key:value pair as string, since otherwise, you could not insert lists or dictionaries as a key
+
+     Parameters
+    ----------
+    data: dict
+        Data dictionary
+    field_name: str
+        Field name to insert
+    field_value: str
+        Field value to insert
+    append_data: dict
+        Additional data to insert
+    """
+    if type(field_name) is str:
+        data[field_name] = field_value
+        string_data = json.dumps(data, indent=4, ensure_ascii=False)
+    else:
+        data['replace_me'] = field_value
+        string_data = json.dumps(data, indent=4, ensure_ascii=False).replace('"replace_me"', f"{field_name}")
+
+    string_data += f",\n{json.dumps(append_data, indent=4, ensure_ascii=False)}"
+
+    string_data = f"[\n{string_data}\n]"
+
+    return string_data
```

Comment:
We can simplify this code by changing it to this:
```suggestion
    data[f'{field_name}'] = field_value
    raw_data = json.dumps(data, indent=4, ensure_ascii=False)
    
    return f"[\n{raw_data},\n{json.dumps(append_data, indent=4, ensure_ascii=False)}\n]"
```

Reply:
Done in ba7bcc0729d9649d5d646f0d5ea80a440267f65b